frontend/src/pages/offers/CreatePalletModal.tsx

```diff
@@ -0,0 +1,106 @@
+import { FormEvent, FunctionComponent, useEffect, useState } from 'react'
+import Button from '../../components/Button'
+import Label from '../../components/forms/Label'
+import { CloseReason, ModalBaseProps } from '../../components/model/ModalBase'
+import PlainModal from '../../components/model/PlainModal'
+import { PALLET_CONFIGS } from '../../data/constants'
+import { PalletType } from '../../types/api-types'
+import PalletCard from './PalletCard'
+
+interface Props extends Pick<ModalBaseProps, 'onRequestClose' | 'isOpen'> {
+  /**
+   * Callback triggered by submitting the form in the modal
+   */
+  onCreatePallet: (palletType: PalletType) => void
+  /**
+   * If true, the controls in the modal will stop responding to user input
+   */
+  disabled?: boolean
+}
+
+/**
+ * Allow users to create a new pallet for a specific offer.
+ */
+const CreatePalletModal: FunctionComponent<Props> = (props) => {
```

Comment:
This thing:

<img width="828" alt="Screen Shot 2021-03-22 at 7 58 49 PM" src="https://user-images.githubusercontent.com/3411183/112086219-0fddce00-8b49-11eb-82d0-d64c36ed3c4d.png">